openforcefield/topology/molecule.py

```diff
@@ -1888,64 +1887,162 @@ def from_smiles(smiles,
+                else:
+                    is_equal = None
+                if bond_stereochemistry_matching:
+                    if is_equal is None:
+                        is_equal = (x['stereochemistry'] == y['stereochemistry'])
+                    else:
+                        is_equal &= (x['stereochemistry'] == y['stereochemistry'])
+
+                return is_equal
+        else:
+            edge_match_func = None
+
+        # Here we should work out what data type we have, also deal with lists?
+        def find_data_type(data):
+            """Find the data type and return the networkx graph"""
+
+            try:
+                # Molecule class instance
+                return data.to_networkx()
+            except AttributeError:
+                try:
+                    # TopologyMolecule class instance
+                    return data.reference_molecule.to_networkx()
+                except AttributeError:
+                    if isinstance(data, nx.Graph):
+                        return data
+
+        mol1_netx = find_data_type(mol1)
+        mol2_netx = find_data_type(mol2)
+        isomorphic = nx.is_isomorphic(mol1_netx,
+                                      mol2_netx,
+                                      node_match=node_match_func,
+                                      edge_match=edge_match_func)
+
+        if isomorphic and return_atom_map:
+            # now generate the sorted mapping between the molecules
+            GM = GraphMatcher(
+                mol1_netx,
+                mol2_netx,
+                node_match=node_match_func,
+                edge_match=edge_match_func)
+            for mapping in GM.isomorphisms_iter():
+                topology_atom_map = mapping
+                break
+
+            # reorder the mapping by keys
+            sorted_mapping = {}
+            for key in sorted(topology_atom_map.keys()):
+                sorted_mapping[key] = topology_atom_map[key]
+
+            return isomorphic, sorted_mapping
 
-        return nx.is_isomorphic(self.to_networkx(),
-                                other.to_networkx(),
-                                node_match=node_match_func,
-                                edge_match=edge_match_func
-                                )
-        #if not (isinstance(other, FrozenMolecule)):
+        else:
+            return isomorphic, None
+
+        # if not (isinstance(other, FrozenMolecule)):
         #    other_fm = FrozenMolecule(other)
-        #else:
+        # else:
         #    other_fm = other
-        #self_smiles = self.to_smiles(toolkit_registry=toolkit_registry)
-        #other_smiles = other_fm.to_smiles(toolkit_registry=toolkit_registry)
-        #return self_smiles == other_smiles
+        # self_smiles = self.to_smiles(toolkit_registry=toolkit_registry)
+        # other_smiles = other_fm.to_smiles(toolkit_registry=toolkit_registry)
+        # return self_smiles == other_smiles
+
+    def is_isomorphic_with(self, other):
```

Comment:
I'd recommend adding **kwargs here, to pass on to the underlying staticmethod.

Reply:
And also copying the docstring for all the additional keyword arguments that can be accepted by `are_isomorphic` into this function's docstring as well.

Reply:
Good idea I have added this in now.